api/scpca_portal/models/library.py

```diff
@@ -91,13 +91,13 @@ def get_data_file_paths(cls, data: Dict, s3_input_bucket: str) -> List[Path]:
         library_id = data.get("scpca_library_id")
         relative_path = Path(f"{project_id}/{sample_id}/{library_id}")
 
-        data_file_paths = [
-            file_path
-            for file_path in s3.list_input_paths(relative_path, s3_input_bucket)
-            if "metadata" not in file_path.name
-        ]
+        file_paths = s3.list_input_paths(relative_path, s3_input_bucket)
+
+        # input metadata json is included in spatial downloads, but excluded from all others
+        if Library.get_modality_from_file_paths(file_paths) == Library.Modalities.SPATIAL:
+            return file_paths
```

Comment:
```suggestion
        # input metadata json is excluded from single_cell downloads
        if Library.get_modality_from_file_paths(file_paths) == Library.Modalities.SINGLE_CELL:
            return [file_path for file_path in file_paths if "metadata" not in file_path.name]
```

Inverting the exception here would make it easier to maintain. This allows you to pair the logic for the exception and the return value in the same statement.